From: Elia Pinto (devzero2000) <pinto.elia@gmail.com>
Subject: libpopt-dev: __secure_getenv() function renamed to secure_getenv()

Use secure_getenv if it is available.
Fall back to __secure_getenv if it is not.
In glibc 2.17, __secure_getenv was renamed to secure_getenv.
So check also for it.
Reference:
 http://sourceware.org/glibc/wiki/Tips_and_Tricks/secure_getenv
 http://marc.info/?t=134192618500001&r=1&w=2

Origin: upstream, http://rpm5.org/cvs/chngview?cn=16984
Bug-Debian: http://bugs.debian.org/728596


Index: popt-1.16/configure.ac
===================================================================
--- popt-1.16.orig/configure.ac	2010-05-04 21:55:54.000000000 +0100
+++ popt-1.16/configure.ac	2013-11-03 19:54:39.838667371 +0000
@@ -82,7 +82,7 @@
 AC_CHECK_FUNC(setreuid, [], [
     AC_CHECK_LIB(ucb, setreuid, [if echo $LIBS | grep -- -lucb >/dev/null ;then :; else LIBS="$LIBS -lc -lucb" USEUCB=y;fi])
 ])
-AC_CHECK_FUNCS(getuid geteuid iconv mtrace __secure_getenv setregid stpcpy strerror vasprintf srandom)
+AC_CHECK_FUNCS(getuid geteuid iconv mtrace __secure_getenv secure_getenv setregid stpcpy strerror vasprintf srandom)
 
 AM_GNU_GETTEXT([external])
 AM_ICONV_LINK
Index: popt-1.16/system.h
===================================================================
--- popt-1.16.orig/system.h	2008-12-18 17:09:49.000000000 +0000
+++ popt-1.16/system.h	2013-11-03 20:06:58.305755167 +0000
@@ -91,9 +91,13 @@
 #define	xstrdup(_str)	strdup(_str)
 #endif  /* defined(HAVE_MCHECK_H) && defined(__GNUC__) */
 
-#if defined(HAVE___SECURE_GETENV) && !defined(__LCLINT__)
+#if !defined(__LCLINT__)
+#if defined(HAVE_SECURE_GETENV)
+#define	getenv(_s)	secure_getenv(_s)
+#elif defined(HAVE___SECURE_GETENV)
 #define	getenv(_s)	__secure_getenv(_s)
 #endif
+#endif
 
 #if !defined(__GNUC__) && !defined(__attribute__)
 #define __attribute__(x) 
